core/src/main/scala/org/apache/spark/ui/ToolTips.scala

```diff
@@ -62,6 +62,13 @@ private[spark] object ToolTips {
     """Time that the executor spent paused for Java garbage collection while the task was
        running."""
 
+  val PEAK_EXECUTION_MEMORY =
+    """Execution memory refers to the memory used by internal data structures created during
+       shuffles, aggregations and joins when Tungsten is enabled. The value of this accumulator
+       should be approximately the sum of the peak sizes across all such data structures created
+       in this task. For SQL jobs, this only tracks all unsafe operators, broadcast joins, and
+       external sort."""
```

Comment:
I don't love the name "Execution Memory" -- I think a user will assume this is covering *all* memory that isn't for cached rdds.  I don't even think that is 100% accurate for sql, but it'll be even less accurate for non-sql use.

Not that I have much better suggestions ... "Peak Aggregator Memory" maybe?